src/TNetwork.cpp

```diff
@@ -196,10 +253,16 @@ void TNetwork::Identify(TConnection&& RawConnection) {
         RawConnection.Socket.shutdown(socket_base::shutdown_both, ec);
         return;
     }
+    std::string clientAddress = RawConnection.SockAddr.address().to_string();
     std::shared_ptr<TClient> Client { nullptr };
+    WatchingConnecting connectionManager;
     try {
-        if (Code == 'C') {
-            Client = Authentication(std::move(RawConnection));
+        if (Code == 'C') { 
+             if (connectionManager.IsConnectionAllowed(clientAddress)) {
+                Client = Authentication(std::move(RawConnection));
+             } else {
+                RawConnection.Socket.shutdown(socket_base::shutdown_both, ec);
```

Comment:
Could we maybe send a message here, and then close? It will be non-trivial to debug if this happens to someone by accident, they will think BeamMP is unstable.